### DISCUSSION TITLE
Trouble with getting Lesson 13 NFT (from Advanced Foundry Section 5)

### DISCUSSION THREAD

**Discussion by Joro97** [Cyfrin/foundry-full-course-cu#2795] [Q&A] [answered]
I have written [this smart contract](https://sepolia.etherscan.io/address/0x49794618a14444f1449698548f519139e94a80df#code) as a solution to lesson 13 challenge, however it does not seem to work and I cant understand why. [I call the solve function](https://sepolia.etherscan.io/tx/0x24b360210a61b2dc29595d975f0308b33f4b3ffe1ddf4c1849275240e10c19c1) with the same address as the one I have created the solution contract with and have validated in Remix that my number should be correct, but I am still getting my transactions reverted with the `Custom Error 'LessonThirteen__NotSolved ()`, can somebody see what I am doing wrong?

**Comment by Joro97** [comment]
Also as a small improvement maybe the challenge custom errors could be more granular, becoming something like: 
```
    error LessonThirteen__WrongSender();
    error LessonThirteen__CouldNotDelegateCall();
    error LessonThirteen__WrongNumber(uint256); 
```
so that a person can know exactly what is wrong with their submission. If someone from Cyfrin agrees that this would make sense I can open a PR with the change.

**Comment by Joro97** [comment]
I have also tried some basic things like changing my solidity complier version to match whats in the challenge contract, not having the interface as part of my contract and similar, but still cant get a successful transaction.

**Comment by EngrPips** [comment]
Hello @Joro97, That is why it is a challenge, and you need to figure out how to solve it. One hint I can give you is to make sure you read the challenge contract and understand what is required of you.

**Reply by Joro97** [reply]
@EngrPips well I have been trying to do this for a few hours 😄 , but cant find whats wrong with my submission. My logic is that when calling `address(i_helperContract).delegatecall(abi.encodeWithSignature("addTen(int256)", number)); (LessonThirteenHelper)` it would use the number from my contract - 1325 as and argument to the function, then add what is in the first storage slot of the caller of the delegate call (), which should be` LESSON_VERSION = 2` and add 10 from the addTen function. When deploying my solution contract + LessonThirtheen and LessonThirteenHelper in Remix and passing the local chain address of the solution contract - it reaches the target number.

**Reply by EngrPips** [reply]
Challenges sometimes take days and sometimes weeks, so that is completely normal. One more tip is to check for any successful attempt to solve the challenge and see what you can learn from the transaction that may help you.

**Reply by Joro97** [reply]
Yeah, I tired this and initially could not find a solution contract that was verified, but I already found a verified one and am looking for some more clues on what am I missing, thanks for the tips!

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.

**Comment by Joro97** [comment, marked as answer]
Managed to solve it with the additional hint from a friend and I understood what my error was. Pretty interesting task. I have also left a small hint in my solution in case someone else is stuck and stumbles upon it. Thanks for your help @EngrPips!

**Reply by EngrPips** [reply]
Anytime, Friend. Keep crushing it.
